JumpscaleCore/clients/test/test_sshkey_client.py

```diff
@@ -0,0 +1,134 @@
+        """
+        TC
+        Test case for private key method on sshkey client
+
+        **Test scenario**
+        #. compare the output from this method with the contact of /root/.ssh/sshkey_client
+        """
+        self.info("check privkey method in sshkey client")
+        ssh_key_privkey = self.sshkey_client.privkey
+        self.assertEqual(ssh_key_privkey, open('/root/.ssh/sshkey_client').read())
+        return ssh_key_privkey
+
+    def test003_name(self):
+        """
+        TC
+        Test case for name method of sshkey client
+
+        **Test scenario**
+        #. check the name of sshkey client it must equal to sshkey_client
+        """
+        self.info("check name method in sshkey client")
+        ssh_key_name = self.sshkey_client.name
+        self.assertEqual(ssh_key_name, "sshkey_client")
+
+    def test004_path(self):
+        """
+        TC
+        Test case for path of sshkey client
+
+        **Test scenario**
+        #. check the path of sshkey client it must equal to /root/.ssh/sshkey_client
+        """
+        self.info("check path method in sshkey client, it should equal to /root/.ssh/sshkey_client")
+        ssh_key_path = self.sshkey_client.path
+        self.assertEqual(ssh_key_path, '/root/.ssh/sshkey_client')
+
+    def test005_pubkey_only(self):
+        """
+        TC
+        Test case for pubkey_only method in sshkey client
+
+        **Test scenario**
+        #. check the pubkey_only for sshkey client
+        """
+        self.info("check pubkey_only method in sshkey client")
+        ssh_key_pubkey_only = self.sshkey_client.pubkey_only
+        pubkey_only = open('/root/.ssh/sshkey_client').read().split()[1]
+        self.assertEqual(ssh_key_pubkey_only, pubkey_only)
+
+    def test006_delete_from_sshdir(self):
+        """
+        TC
+        Test case for delete_from_sshdir method in sshkey client
+
+        **Test scenario**
+        #. use delete_from_sshdir to delete the sshkey_client files from ssh directory
+        #. check the existence of those files (sshkey_client & sshkey_client.pub) in the sshkey directory.
+        """
+        self.info("check delete_from_sshdir method in sshkey client")
+        self.sshkey_client.delete_from_sshdir()
+        self.info("check that sshkey_client files are deleted from ssh directory")
+        self.assertFalse(os.path.isfile('/root/.ssh/sshkey_client'))
+        self.assertFalse(os.path.isfile('/root/.ssh/sshkey_client.pub'))
+
+    def test007_write_to_sshdir(self):
+        """
+        TC
+        Test case for write_to_sshdir method in sshkey client
+
+        **Test scenario**
+        #. use write_to_sshdir method to write sshkey again to the directory.
+        #. check the existence of those files (sshkey_client & sshkey_client.pub) in the sshkey directory.
+        #. check the public and private keys should be the same as before
+        """
+        self.info("check write_to_sshdir method in sshkey client")
```

Comment:
tests should be separate as in this scenario u depend on that key deleted before in  test006